src/redux/books/booksSlice.js

```diff
@@ -1,43 +1,95 @@
-import { createSlice } from '@reduxjs/toolkit';
+/* eslint-disable no-param-reassign */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const API_BOOKS_URL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/0AXvckrVNcvHP5pgzcwk/books/';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: {},
+  ifSucceed: false,
+  isLoading: false,
+  error: null,
+};
+
+const headers = {
+  headers: {
+    'content-type': 'application/json',
+  },
 };
 
+const fetchBooks = createAsyncThunk(
+  'books/fetchBooks',
+  async () => {
+    try {
+      const { data } = await axios.get(API_BOOKS_URL);
+      return data;
+    } catch (err) {
+      return err.message;
+    }
+  },
+);
+
+const addBook = createAsyncThunk('books/addBooks', async (book) => {
+  const data = JSON.stringify(book);
+  try {
+    const response = await axios.post(API_BOOKS_URL, data, headers);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
+const removeBook = createAsyncThunk('books/removeBook', async (id) => {
+  const data = JSON.stringify({ item_id: id });
+  try {
+    const response = await axios.delete(API_BOOKS_URL + id, data, headers);
+    return response.data;
+  } catch (error) {
+    return error;
+  }
+});
+
 const booksSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {
-    addBook: (state, action) => ({
-      ...state,
-      books: [...state.books, action.payload],
-    }),
-    removeBook: (state, action) => ({
-      ...state,
-      books: state.books.filter((book) => book.item_id !== action.payload),
-    }),
+  reducers: {},
+  extraReducers: {
+    [fetchBooks.pending]: (state) => {
+      state.status = true;
+    },
+    [fetchBooks.fulfilled]: (state, action) => {
+      state.isLoading = false;
+      state.ifSucceed = true;
+      state.books = action.payload;
+    },
+    [fetchBooks.rejected]: (state) => {
+      state.isLoading = false;
+    },
+
+    [addBook.pending]: (state) => {
+      state.isLoading = true;
+    },
+    [addBook.fulfilled]: (state) => {
+      state.isLoading = false;
+      state.ifSucceed = false;
+    },
+    [addBook.rejected]: (state) => {
+      state.isLoading = false;
+      state.ifSucceed = false;
+    },
+  },
+
+  [removeBook.pending]: (state) => {
+    state.isLoading = true;
+  },
+  [removeBook.fulfilled]: (state) => {
+    state.isLoading = false;
+  },
+  [removeBook.rejected]: (state) => {
+    state.isLoading = false;
```

Comment:
- [ ] Kindly ensure that the state is updated after removing books so that the removed book will disappear from the list. Currently, the removed book only disappears after refresh.

Reply:
Thank you